ray-operator/controllers/ray/rayservice_controller_unit_test.go

```diff
@@ -1153,10 +1153,88 @@ func TestShouldPrepareNewCluster_ZeroDowntimeUpgrade(t *testing.T) {
 
 	// Update cluster spec in RayService to trigger a zero downtime upgrade.
 	rayService.Spec.RayClusterSpec.RayVersion = "new-version"
-	shouldPrepareNewCluster := shouldPrepareNewCluster(ctx, &rayService, activeCluster, nil)
+	shouldPrepareNewCluster := shouldPrepareNewCluster(ctx, &rayService, activeCluster, nil, false)
 	assert.True(t, shouldPrepareNewCluster)
 }
 
+func TestShouldPrepareNewCluster_RecreatePendingCluster(t *testing.T) {
```

Comment:
These two tests look pretty similar how about merging it into one test with `t.Run(...)`.

Reply:
Done.